api/bin/chainflip-lp-api/src/main.rs

```diff
@@ -601,13 +622,20 @@ impl RpcServer for RpcServerImpl {
 
 	async fn cancel_orders_batch(
 		&self,
-		orders: BoundedVec<CloseOrder, ConstU32<MAX_ORDERS_DELETE>>,
+		orders: BoundedVec<CloseOrderJson, ConstU32<MAX_ORDERS_DELETE>>,
 		wait_for: Option<WaitFor>,
 	) -> RpcResult<ApiWaitForResult<Vec<LimitOrRangeOrder>>> {
+		let mut final_orders: BoundedVec<CloseOrder, ConstU32<MAX_ORDERS_DELETE>> =
+			BoundedVec::new();
+		for order in orders {
+			final_orders
+				.try_push(order.try_into()?)
+				.expect("Impossible to fail, given the same MAX_ORDERS_DELETE");
+		}
```

Comment:
can remove the mut by collecting
```suggestion
Ok(self
			.api
			.lp_api()
			.cancel_orders_batch(
				orders
					.into_iter()
					.map(TryInto::try_into)
					.collect::<Result<Vec<_>, _>>()?
					.try_into()
					.expect("Impossible to fail, given the same MAX_ORDERS_DELETE"),
				wait_for.unwrap_or_default(),
			)
			.await?)
```

Reply:
Doesn't this require an extra memory allocation for the `Vec<>` when collecting?
(or maybe this gets optimized away by the compiler?)

Reply:
Yes, I believe so. But given the performance gain is negligible, I think having without a mut var is a bit simpler and less likely to be mistakenly broken with future changes (definitely isn't a strong opinion which way is preferred here)